go-runtime/compile/schema.go

```diff
@@ -223,6 +252,114 @@ func goPosToSchemaPos(pos token.Pos) schema.Position {
 	return schema.Position{Filename: p.Filename, Line: p.Line, Column: p.Column, Offset: p.Offset}
 }
 
+func visitGenDecl(pctx *parseContext, node *ast.GenDecl) error {
+	switch node.Tok {
+	case token.TYPE:
+		if node.Doc == nil {
+			return nil
+		}
+		directives, err := parseDirectives(fset, node.Doc)
+		if err != nil {
+			return err
+		}
+		for _, dir := range directives {
+			switch dir.(type) {
+			case *directiveEnum:
+				enum := &schema.Enum{
+					Pos:      goPosToSchemaPos(node.Pos()),
+					Comments: visitComments(node.Doc),
+				}
+				if len(node.Specs) != 1 {
+					return fmt.Errorf("%s: error parsing ftl enum %s: expected exactly one type spec",
+						goPosToSchemaPos(node.Pos()), enum.Name)
+				}
+				if t, ok := node.Specs[0].(*ast.TypeSpec); ok {
+					pctx.enums[t.Name.Name] = enum
+					err := visitTypeSpec(pctx, t)
+					if err != nil {
+						return err
+					}
+				}
+			case *directiveIngress, *directiveModule, *directiveVerb:
+			}
+		}
+		return nil
+	case token.CONST:
+		var typ ast.Expr
+		for i, s := range node.Specs {
+			if v, ok := s.(*ast.ValueSpec); ok {
+				// In a simple enum, only the first value has a type.
+				if i == 0 && isSimpleEnumIdent(v) {
```

Comment:
Is this referring to `iota` enums? eg.

```go
//ftl:enum
type Foo int

const (
  One Foo = iota
  Two
  Three
)
```

And if not, will `iota` enums work?

Reply:
renamed for clarity - per other discussion iota enums (and expressions) are working!